nnfabrik/examples/mnist/trainer.py

```diff
@@ -13,31 +14,42 @@ def __init__(self, model, dataloaders, seed, epochs=5):
         self.trainloader = dataloaders["train"]
         self.seed = seed
         self.epochs = epochs
-        self.loss_fn = nn.MSELoss()
+        self.loss_fn = nn.NLLLoss()
         self.optimizer = optim.Adam(self.model.parameters())
 
     def train(self):
+        if hasattr(tqdm, "_instances"):
+            tqdm._instances.clear()  # To have tqdm output without line-breaks between steps
         torch.manual_seed(self.seed)
-        losses = []
+        accs = []
         for epoch in range(self.epochs):
-
-            _losses = []
-            for x, y in self.trainloader:
-
+            predicted_correct = 0
+            total = 0
+            for x, y in tqdm(self.trainloader):
+                # forward:
                 self.optimizer.zero_grad()
-                y_hat = self.model(x)
+                x_flat = x.flatten(1, -1)  # treat the images as flat vectors
+                y_hat = self.model(x_flat)
                 loss = self.loss_fn(y_hat, y)
+                # backward:
                 loss.backward()
                 self.optimizer.step()
-                _losses.append(loss.item())
-
-        losses.append(np.mean(_losses))
+                # keep track of accuracy:
```

Comment:
I think the code is self-explanatory and the comment is not really needed.

Reply:
This is meant more to structure the code into forward, backward and tracking.

Reply:
I see. In that case I would personally prefer having separate (private) functions for forward backward and tracking instead of commented sections. That said I am also fine with it staying as it is.

Reply:
That's again a trade-off between good practices and compact code. I opted for the latter here.